### ISSUE TITLE
Infinite scroll issue when switching tabs

### ISSUE THREAD

**Issue by nonybrighto** [kserko/CineReel#3]
Thanks a lot for this work! From the comment in the bloc_provider.class file, infinite scroll doesn't work when switching tabs. Did you get an alternative approach to this? It works fine if the streams are prevented from closing, but I am hoping for something neater.

**Comment by kserko** [comment]
No I haven't had time to take a closer look at this yet, but thanks for bringing it up, I need to prioritize it and see what the best solution is

**Comment by nonybrighto** [comment]
ok. Thanks!

**Comment by whataa** [comment]
Let `BlocProvider` mixin `AutomaticKeepAliveClientMixin` and provide a bool value to switch it.

Sorry I have no key to run and test, so I don't know if this will solve it.

**Comment by kserko** [comment]
So I think I got a fix for this. The main changes are 

[line 32](https://github.com/kserko/CineReel/commit/24be0aebeeb46ad66ed241eb5bdd24b8814e6639#diff-0302503593cd45ce435345ee98b5a699R32) in the **movie_bloc.dart** where we check if the streamController is closed (i.e we navigated away from a tab and the controller got disposed) and if so, we re-instantiate it before returning the stream object 

and

[line 49](https://github.com/kserko/CineReel/commit/24be0aebeeb46ad66ed241eb5bdd24b8814e6639#diff-75b738ee2712eddfd112d45e3dc93cb4R49) in **movies_list_screen.dart** where we provide the StreamBuilder with initialData. This is important because if we don't do this, then if we go away and then come back into a tab this would cause the stream to be disposed at first (when we left the tab) and the recreated again (as described above) but it would also mean that the new stream wouldn't yet have any data added to it, so the StreamBuilder wouldn't build anything and our list would be empty. By passing initialData, the list always has some data to start with, and then the stream takes over in providing the updates to the build method as we scroll down.

I hope that makes some sense. I'm not sure this is the most elegant solution but it looks like it works.
